test/widgets/message_list_test.dart

```diff
@@ -923,6 +923,53 @@ void main() {
         await tester.pump();
         tester.widget(find.text('new stream name'));
       });
+
+      testWidgets('navigates to TopicNarrow on tapping topic in ChannelNarrow', (tester) async {
+        final pushedRoutes = <Route<void>>[];
+        final navObserver = TestNavigatorObserver()
+          ..onPushed = (route, prevRoute) => pushedRoutes.add(route);
+        final channel = eg.stream();
+        await setupMessageListPage(tester,
+          narrow: ChannelNarrow(channel.streamId),
+          messages: [eg.streamMessage(stream: channel, topic: 'topic name')],
+          streams: [channel],
+          navObservers: [navObserver]);
+
+        assert(pushedRoutes.length == 1);
+        pushedRoutes.clear();
+
+        connection.prepare(json: eg.newestGetMessagesResult(
+          foundOldest: true, messages: [message]).toJson());
+        await tester.tap(find.descendant(
+          of: find.byType(StreamMessageRecipientHeader),
+          matching: find.text('topic name')));
+        await tester.pumpAndSettle();
+        check(pushedRoutes).single
+          .isA<WidgetRoute>()
+          .page.isA<MessageListPage>()
+          .initNarrow.equals(TopicNarrow(channel.streamId, TopicName('topic name')));
+      });
+
+      testWidgets('does not navigate on tapping topic in TopicNarrow', (tester) async {
+        final pushedRoutes = <Route<void>>[];
+        final navObserver = TestNavigatorObserver()
+          ..onPushed = (route, prevRoute) => pushedRoutes.add(route);
+        final channel = eg.stream();
+        await setupMessageListPage(tester,
+          narrow: TopicNarrow(channel.streamId, TopicName('topic name')),
+          navObservers: [navObserver],
+          streams: [channel],
+          messages: [eg.streamMessage(stream: channel, topic: 'topic name')]);
```

Comment:
bump https://github.com/zulip/zulip-flutter/pull/1193#discussion_r1926378103

This one has navObservers before streams and messages, while the other test has it after.